### PR TITLE
Rustup to rustc 1.71.1

### PR DESCRIPTION
**Pull request by bjorn3**

### CODE REVIEW

File: source/rust_verify/src/rust_to_vir_expr.rs [hunk 1]
```diff
@@ -1286,6 +1286,9 @@ pub(crate) fn expr_to_vir_innermost<'tcx>(
         ExprKind::AddrOf(BorrowKind::Raw, _, _) => {
             unsupported_err!(expr.span, format!("raw borrows"))
         }
+        ExprKind::OffsetOf(_container, _fields) => {
+            unsupported_err!(expr.span, format!("offset_of!()"))
```
Comment (bjorn3) [inline source/rust_verify/src/rust_to_vir_expr.rs:1290]: I don't think this is implementable at the HIR level. It needs to know the layout of a monomorphized type.

Comment (utaal) [reply source/rust_verify/src/rust_to_vir_expr.rs:1290]: Thanks for the note. I don't think it's a priority for Verus.

**Comment by bjorn3** [comment]
Could you please either point me to where the unused variables the tests trip over are generated or fix them for me?

**Comment by utaal** [comment]
Hey. Thanks for the PR. I don't immediately know what's causing the failures. I'll take a look as soon as I can.

**Review by utaal — APPROVED** [review]

**Comment by utaal** [comment]
Sorry for the (long) delay. And thanks again!

**Comment by bjorn3** [comment]
No problem. I don't plan on doing the rustup to 1.72.1 or later though as I'm busy with fixing a whole lot of bug reports to cg_clif. So you will have to do that yourself at some point.

**Comment by utaal** [comment]
Thank you for all your contributions so far! (The cranelift backend looks very cool!)

**Pull request merged**